### ISSUE TITLE
Convention: `environment` attribute (production, staging...)

### ISSUE THREAD

**Issue by cyrille-leclerc** [open-telemetry/opentelemetry-specification#752]
**What are you trying to achieve?**

Context: Many Observability system collect observability data coming from multiple environments (production, staging...).

What I try to achieve is to characterize every telemetry data indicating emitting environment (`production`, `staging`, `dev`...).

These telemetry data can be data of:
* any type: traces, logs, metrics or synthetics
* any layer: application layer, infrastructure layer...

Specifying an environment seems to be relevant to characterise the following entities:
*  Service:
   * a service deployment used for production workloads
* Kubernetes:
   * pod / deployment: e.g. the deployment of an application used for "production" workloads
   * cluster: e.g. a Kubernetes cluster dedicated to "production"
* Host: e.g. a server used for "production" workloads

Specifying the "environment" on the "Service" level is immediately useful for OpenTelemetry traces (ie the application layer) as traces are already declared stable with many Observability solution supporting them.
We should also anticipate the importance of characterizing "hosts" and probably "kubernetes deployments/clusters"  as soon as OpenTelemetry is stable for metrics and logs.

At Elastic, we discussed of whether this "environment" attribute should be multi valued or mono valued. We tend to think that the adoption is easier with a mono valued attribute and that if a resource is used for multiple environments, then it could be associated with the most critical environment (e.g. production).
For example, for support purpose and security audit purpose, it is very important to understand if a resource is used for 1 production component. The fact that there would also be non production workloads would not lower the criticality of the observed resource.

**Example using the [Google Online Boutique Demo](https://github.com/GoogleCloudPlatform/microservices-demo)**

Assuming that my observability platform is connected to both "staging" and "production" environments, when I investigate on my "Online Boutique" system and particularly the "Checkout Service", I want to filter data so that I only see the data related to the "production" environment, excluding the data related to the "staging" or "dev" environment.

Sample of attributes collected that could be collected today for a span of, showing that the "environment" attribute could be used to characterise multiple entities : service deployment, kubernetes deployment and host.

```
# SPAN / TRACE
# Pseudo attributes, I didn't find a "semantic convention" for span_id, span_name and trace_id
trace.span_id: 7a2d30add01713e6
trace.span_name: checkout
trace.trace_id: 3c8ec699af7735ba
...

# SERVICE
# https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/README.md#service
service.name: checkoutservice
service.namespace: onlineboutique
service.version: v0.2.0
service.instance.id: 627cc493-f310-47de-96bd-71410b7dec09

# CONTAINER
# https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/container.md
container.name: onlineboutique-checkoutservice
container.id: a3bf90e006b2
container.image.name: mycompany.com/onlineboutique/checkoutservice
container.image.tag: v0.2.0.3

#KUBERNETES
# https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/k8s.md
k8s.pod.uid: 275ecb36-5aa8-4c2a-9c47-d8bb681b9aff
k8s.pod.name: onlineboutique-pod
k8s.cluster.name: ecommerce
...

# HOST
# https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/host.md
host.name: my-kubernetes-node-123
...

# NET
# https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes
net.host.ip:10.0.0.5
```

**Additional context.**

FYI Elastic is also working at the moment at standardissing an environment attribute in the Elastic Common Schema: https://github.com/elastic/ecs/blob/master/rfcs/text/0002-rfc-environment.md

**Comment by cyrille-leclerc** [comment]
Note: the label "spec:resource" is restrictive, this issue probably deserves the labels "area:semantic-conventions" and "spec:trace"

**Comment by arminru** [comment]
Hi @cyrille-leclerc!
PR #606, which has been around for a while now but unfortunately neglected by reviewers so far, intends to resolve the related issue #593 by adding a deployment environment resource attribute.
Would that one solve your needs or would you like to add anything to that one?

**Comment by Oberon00** [comment]
> but unfortunately neglected by reviewers so far

I disagree. Reviewers agreed that the name needs to be changed from `environment` to something less ambiguous, multiple suggestions have been made. IMHO the ball is in the court of PR author @ecourreges-orange there.

**Comment by cyrille-leclerc** [comment]
Thanks @arminru and @Oberon00 . My issue is definitively a duplicate. I hope the research I have made can help move forward.

**Comment by webmat** [comment]
@cyrille-leclerc Great additional resources, thanks for dropping these notes here.

When do you think you can open the PR for the next stage?
